mining.md

```diff
@@ -119,17 +119,27 @@ func VerifyBlock(blk Block) {
         Fatal("invalid block signature")
     }
     
-    // 2. Verify ParentWeight
+    // 2. Verify Timestamp
+    // first check that it is not in the future
+    if blk.GetTime() > nAdjustedTime {
+        Fatal("block was generated too far in the future")
+    }
+    // next check that it is appropriately delayed from its parents
+    if blk.GetTime() <= blk.maxParentTime() + BLOCK_DELAY {
```

Comment:
Hrm... I think its probably fine to use minParentTime here

Reply:
Can you discuss pros and cons here?

It's strictly safer to use maxParentTime, though maybe it incentivizes people to not include all the blocks they hear about...

minParentTime doesn't have this issue, but could lead folks to include invalid future blocks in their TipSet that wouldn't get caught? (I'm not fully paged in at this point) which would be a deal breaker...

Reply:
> it incentivizes people to not include all the blocks they hear about...

Right, I think this incentive is a bit weird, and we should avoid it if possible.

Also note: We don't have to restart block generation if we hear about a new parent that doesnt have a smaller ticket, its harmless to include it in our parent set while we're running the vdf.

>  but could lead folks to include invalid future blocks in their TipSet that wouldn't get caught?

I'm not sure I follow here, what exactly could happen?

Reply:
nvm. I'm not sure where I was going with this. Would you also include some margin of error as BTC does, like
` minTime + BLOCK_DELAY + MARGIN` or no. 

I tend to no. We can fix in testnet if needed.

Reply:
Sorry, I've been completely confused about this. Please disregard all other comments (save the one about margin perhaps)

`maxParentTime` gives the miner more time to include new blocks. I expect it's better than `minParentTime` which might be too stringent a rule (in case of network delay eg).